src/ducks/categorization/services.js

```diff
@@ -34,24 +34,24 @@ export const getAlphaParameter = (
   return Math.max(min, Math.min(max, alpha))
 }
 
-const createLocalClassifier = async () => {
-  const getTransWithManualCat = async (transactions, index, limit, skip) => {
-    const query = {
-      selector: { manualCategoryId: { $exists: true } },
-      limit,
-      skip,
-      wholeResponse: true
-    }
-    const results = await cozyClient.data.query(index, query)
-    transactions = [...transactions, ...results.docs]
-
-    if (results.next) {
-      return getTransWithManualCat(transactions, index, limit, skip + limit)
-    }
-
-    return transactions
+const getTransWithManualCat = async (transactions, index, limit, skip) => {
+  const query = {
+    selector: { manualCategoryId: { $exists: true } },
+    limit,
+    skip,
+    wholeResponse: true
+  }
+  const results = await cozyClient.data.query(index, query)
+  transactions = [...transactions, ...results.docs]
+
+  if (results.next) {
+    return getTransWithManualCat(transactions, index, limit, skip + limit)
   }
 
+  return transactions
+}
```

Comment:
Nit: It's a bit cumbersome to have to call the function with an empty array at first. It can be useful to have a function that does this automatically.

```
const myCumbersomeRecursiveFunction = (res, i) => {
  if (res.length < 100) {
    res.push(i)
    return myCumbersomeRecursiveFunction(res, i+1)
  } else {
    return res
  }
}

const recursiveFunction = i => {
  return myCumbersomeRecursiveFunction({}, i)
} 
```

A default argument is also good but you need to reverse the arguments.